Aquality.WinAppDriver/tests/Aquality.WinAppDriver.Tests/Elements/ElementStateProviderTests.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using Aquality.WinAppDriver.Applications;
+using Aquality.WinAppDriver.Elements.Interfaces;
+using Aquality.WinAppDriver.Tests.Applications.Locators;
+using NUnit.Framework;
+using OpenQA.Selenium;
+
+namespace Aquality.WinAppDriver.Tests.Elements
+{
+    public class ElementStateProviderTests : TestWithApplication
+    {
+        private static readonly CalculatorWindow CalculatorWindow = new CalculatorWindow();
+        private static readonly ITextBox RightArgumentTextBox = CalculatorWindow.RightArgumentTextBox;
+        private static readonly IButton MaximizeButton = CalculatorWindow.MaximizeButton;
+        private static readonly IButton EmptyButton = CalculatorWindow.EmptyButton;
+
+        private IElement notPresentLabel = ApplicationManager.GetRequiredService<IElementFactory>()
+            .GetLabel(By.XPath("//*[@id='111111']"), "Not present element");
+
+        [Test]
+        public void Should_ReturnTrue_IfElementIsDisplayed()
+        {
+            Assert.IsTrue(RightArgumentTextBox.State.IsDisplayed);
+        }
+
+        [Test]
+        public void Should_ReturnFalse_IfElementIsNotDisplayed()
+        {
+            Assert.IsFalse(MaximizeButton.State.IsDisplayed);
+        }
+
+        [Test]
+        public void Should_ReturnFalse_IfElementDoesNotExist()
+        {
+            Assert.IsFalse(notPresentLabel.State.IsExist);
+        }
+
+        [Test]
+        public void Should_ReturnTrue_IfElementExist()
+        {
+            Assert.IsTrue(RightArgumentTextBox.State.IsExist);
+        }
+
+        [Test]
+        public void Should_ReturnFalse_IfElementIsNotClickable()
+        {
+            Assert.IsFalse(EmptyButton.State.IsClickable);
+        }
+
+        [Test]
+        public void Should_ReturnTrue_IfElementIsClickable()
+        {
+            Assert.IsTrue(RightArgumentTextBox.State.IsClickable);
+        }
+
+        [Test]
+        public void Should_ReturnFalse_IfElementIsNotEnabled()
+        {
+            Assert.IsFalse(EmptyButton.State.IsEnabled);
+        }
+
+        [Test]
+        public void Should_ReturnTrue_IfElementIsEnabled()
+        {
+            Assert.IsTrue(RightArgumentTextBox.State.IsEnabled);
+        }
+
+        [Test]
+        public void Should_ThrowException_IfWaitForClickableEnded()
+        {
+            Assert.Throws<WebDriverTimeoutException>(() => EmptyButton.State.WaitForClickable(TimeSpan.Zero));
```

Comment:
@hsuboch , @mialeska don't we want to check that wait lasts expected time? The same question for all wait methods. Or will it be covered or has been already covered in core tests?

Reply:
yes, I believe we should add such tests

Reply:
added